controller/workitem.go

```diff
@@ -233,11 +233,17 @@ func (c *WorkitemController) Reorder(ctx *app.ReorderWorkitemContext) error {
 				return jsonapi.JSONErrorResponse(ctx, errs.Wrap(err, "failed to reorder work item"))
 			}
 
+			// check if the workitems to reorder belongs to the space
+			_, err = appl.WorkItems().Load(ctx, ctx.SpaceID, wi.Number)
```

Comment:
@hectorj2f The idea here is to check that the workitems which are supposed to be reordered belong to the **space** whose id is passed in the payload.  CheckExists() just checks if the workitem is present in the table. It doesnot check if the workitem belongs to a particular space.